src/traversals/bfs.jl

```diff
@@ -116,149 +15,46 @@ function tree(parents::AbstractVector{T}) where T<:Integer
     return t
 end
 
-tree(parents::TreeBFSVisitorVector) = tree(parents.tree)
-
-function examine_neighbor!(visitor::TreeBFSVisitorVector, u::Integer, v::Integer,
-                            ucolor::Int, vcolor::Int, ecolor::Int)
-    if u != v && vcolor == 0
-        visitor.tree[v] = u
-    end
-    return true
-end
-
-function bfs_tree!(visitor::TreeBFSVisitorVector{T},
-        g::AbstractGraph,
-        s::Integer;
-        vertexcolormap = Dict{T,Int}(),
-        queue = Vector{T}()) where T<:Integer
-    # this version of bfs_tree! allows one to reuse the memory necessary to compute the tree
-    # the output is stored in the visitor.tree array whose entries are the vertex id of the
-    # parent of the index. This function checks if the scratch space is too small for the graph.
-    # and throws an error if it is too small.
-    # the source is represented in the output by a fixed point v[root] == root.
-    # this function is considered a performant version of bfs_tree for useful when the parent
-    # array is more helpful than a DiGraph struct, or when performance is critical.
-    nvg = nv(g)
-    length(visitor.tree) >= nvg || error("visitor.tree too small for graph")
-    visitor.tree[s] = s
-    traverse_graph!(g, BreadthFirst(), s, visitor; vertexcolormap=vertexcolormap, queue=queue)
-end
-
-"""
-    bfs_tree(g, s)
-
-Provide a breadth-first traversal of the graph `g` starting with source vertex `s`,
-and return a directed acyclic graph of vertices in the order they were discovered.
-
-### Implementation Notes
-This function is a high level wrapper around [`bfs_tree!`](@ref); use that function for more performance.
-"""
-function bfs_tree(g::AbstractGraph, s::Integer)
-    nvg = nv(g)
-    visitor = TreeBFSVisitorVector(nvg)
-    bfs_tree!(visitor, g, s)
-    return tree(visitor)
-end
-
-############################################
-# Connected Components with BFS            #
-############################################
 """
-    ComponentVisitorVector{T}
+    bfs_parents(g, s[; dir=:out])
 
-A type of `AbstractGraphVisitor` that represents connected components with
-BFS starting from a given seed.
+Perform a breadth-first search of graph `g` starting from vertex `s`.
+Return a vector of parent vertices indexed by vertex. If `dir` is specified,
```

Comment:
I'll see if I can come up with some wording, but my impression is that "parent" is a fairly standard (and well-understood) term. More docs = better, though, so let me mull it over. If it doesn't go in right away it'll go in as a doc change afterwards (doc changes don't require as stringent a review).